### PR TITLE
Support user ldelf for x86 platform

### PR DESCRIPTION
**Pull request by GangSecurity**
Signed-off-by: Chen Gang G <gang.g.chen@intel.com>

### CODE REVIEW

File: ldelf/ldelf.ld.S [hunk 1]
```diff
@@ -4,14 +4,8 @@
  * Copyright (c) 2016-2019, Linaro Limited
```
Comment (jingdlu) [inline ldelf/ldelf.ld.S:4]: Add Intel copyright?

**Review by jingdlu — COMMENTED** [review]
Considering possible upstream requirement, it's better to use macro to distinguish different architecture in common files.

File: ldelf/ldelf.ld.S (hunk 1 above)
Comment (GangSecurity) [reply ldelf/ldelf.ld.S:4]: added

File: ldelf/ldelf.mk [hunk 2]
```diff
@@ -17,6 +17,9 @@ endif
 ifeq ($(CFG_ARM32_core),y)
 CFG_ARM32_$(sm) := y
 endif
+ifeq ($(CFG_X86_64_core),y)
```
Comment (jingdlu) [inline ldelf/ldelf.mk:20]: Where to define CFG_X86_64_core? Don't need this.

File: ldelf/sub.mk [hunk 3]
```diff
@@ -4,6 +4,8 @@ srcs-$(CFG_ARM64_$(sm)) += start_a64.S
 srcs-$(CFG_ARM32_$(sm)) += syscalls_a32.S
 srcs-$(CFG_ARM64_$(sm)) += syscalls_a64.S
 srcs-$(CFG_ARM64_$(sm)) += tlsdesc_rel_a64.S
+srcs-$(CFG_X86_64_$(sm)) += start_64.S
+srcs-$(CFG_X86_64_$(sm)) += syscalls_64.S
```
Comment (jingdlu) [inline ldelf/sub.mk:8]: Suggest to change to the following:
ifeq ($(X86_64),1)
srcs-y += start_64.S
srcs-y += syscalls_64.S
#endif

**Review by jingdlu — APPROVED** [review]

**Review by xyzhao2018 — APPROVED** [review]

**Pull request merged**